run-tests/src/utils.js

```diff
@@ -162,21 +165,75 @@ class TestRunner {
               clearInterval(poller);
               TestRunner._parseApiResult(content);
               this.build_status = content.status;
-              resolve();
+              resolve(content);
             }
           }
         });
       }, WATCH_INTERVAL);
     });
   }
 
+  static async _uploadResults(content) {
+    const promises = [];
+    core.info(`Uploading test report to artifacts for build id: ${content.id}`);
+    const { devices, id: buildId, framework } = content;
+    const rootDir = './reports';
+    const username = process.env[ENV_VARS.BROWSERSTACK_USERNAME].replace("-GitHubAction", "");
+    const accesskey = process.env[ENV_VARS.BROWSERSTACK_ACCESS_KEY];
+    for (const device of devices) {
+      const { sessions } = device;
+      for (const session of sessions) {
+        const { id } = session;
+        const options = {
+          url: `https://${username}:${accesskey}@${URLS.BASE_URL}/${URLS.WATCH_FRAMEWORKS[framework]}/${buildId}/sessions/${id}/${URLS.REPORT[framework]}`,
```

Comment:
How to handle when resultBundle is false

Reply:
This is being handled as part of promise rejection. Server Message will be shown to end user.